google-cloud-trace/lib/google/cloud/trace/time_sampler.rb

```diff
@@ -0,0 +1,99 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Google
+  module Cloud
+    module Trace
+      ##
+      # A sampler determines whether a given request's latency trace should
+      # actually be reported. It is usually not necessary to trace every
+      # request, especially for an application serving heavy traffic. You may
+      # use a sampler to decide, for a given request, whether to report its
+      # trace. A sampler is simply a Proc that takes the Rack environment as an
+      # argument and returns a boolean indicating whether or not to sample the
+      # current request. Alternately, it could be an object that duck-types the
+      # Proc interface by implementing the `call` method.
+      #
+      # TimeSampler is the default sampler. It enforces a certain QPS by
+      # delaying a minimum time between each sample. It also supports
+      # blacklisting certain URI paths that should never be traced.
+      #
+      class TimeSampler
+        ##
+        # Default list of paths for which to disable traces. Currently includes
+        # App Engine Flex health checks.
+        DEFAULT_PATH_BLACKLIST = ["/_ah/health"].freeze
+
+        ##
+        # Create a TimeSampler for the given QPS.
+        #
+        # @param [Number] qps Samples per second. Default is 0.1.
+        # @param [Array{String,Regex}] path_blacklist An array of paths or
+        #     path patterns indicating URIs that should never be traced.
+        #     Default is DEFAULT_PATH_BLACKLIST.
+        #
+        def initialize qps: 0.1, path_blacklist: DEFAULT_PATH_BLACKLIST
+          @delay_secs = 1.0 / qps
+          @last_time = ::Time.now.to_f - @delay_secs
+          @path_blacklist = path_blacklist
```

Comment:
@hxiong388 and @dazuma had a conversation offline about this. Blacklist is functionally a strict subset of sampling, and continuing to treat and implement them separately invites the two features to conflict. However, since path blacklisting is a common case, we should document better _that_ it is happening, and _how_ to configure it. I'll look into that.